core/trino-main/src/main/java/io/trino/execution/executor/TaskExecutor.java

```diff
@@ -187,20 +204,29 @@ public TaskExecutor(
             int guaranteedNumberOfDriversPerTask,
             int maximumNumberOfDriversPerTask,
             VersionEmbedder versionEmbedder,
+            Duration interruptRunawaySplitsTimeout,
+            Predicate<List<StackTraceElement>> interruptibleSplitPredicate,
+            Duration interruptSplitInterval,
             MultilevelSplitQueue splitQueue,
             Ticker ticker)
     {
         checkArgument(runnerThreads > 0, "runnerThreads must be at least 1");
         checkArgument(guaranteedNumberOfDriversPerTask > 0, "guaranteedNumberOfDriversPerTask must be at least 1");
         checkArgument(maximumNumberOfDriversPerTask > 0, "maximumNumberOfDriversPerTask must be at least 1");
         checkArgument(guaranteedNumberOfDriversPerTask <= maximumNumberOfDriversPerTask, "guaranteedNumberOfDriversPerTask cannot be greater than maximumNumberOfDriversPerTask");
+        checkArgument(interruptRunawaySplitsTimeout.getValue(TimeUnit.SECONDS) >= 1.0, "interruptRunawaySplitsTimeout must be at least 1 second");
```

Comment:
is `1s` based off of SPLIT_RUN_QUANTA? if so let's just use that

Reply:
ok.